src/plugins/hetero/src/subgraph_collector.cpp

```diff
@@ -606,6 +609,70 @@ std::pair<ov::hetero::SubgraphsVector, ov::hetero::SubgraphsMappingInfo> ov::het
     return subgraph_collector.run();
 }
 
+void ov::hetero::fix_model_with_paged_attention(std::shared_ptr<ov::Model>& model) {
+    using NodePtr = std::shared_ptr<ov::Node>;
+    std::unordered_set<NodePtr> has_visited_transpose;
+    std::vector<NodePtr> vector_visited_transpose;
+
+    std::function<NodePtr(NodePtr)> find_first_transpose_before_pa = [&](NodePtr root_node) -> NodePtr {
+        auto get_output_node = [](const ov::Output<ov::Node>& output) -> NodePtr {
+            return output.get_node_shared_ptr();
+        };
+        auto get_input_node = [&get_output_node](const ov::Input<ov::Node>& input) -> NodePtr {
+            return get_output_node(input.get_source_output());
+        };
+        auto cur_node = get_input_node(root_node->inputs()[0]);
+        if (ov::is_type<ov::op::v1::Transpose>(cur_node)) {
+            return cur_node;
+        }
+        return find_first_transpose_before_pa(cur_node);
+    };
+
+    auto find_transpose_ops = [&](NodePtr node) {
+        for (size_t i = 0; i < 3; i++) {
+            auto first_transpose_before_pa = find_first_transpose_before_pa(node->get_input_node_shared_ptr(i));
+            if (has_visited_transpose.insert(first_transpose_before_pa).second) {
+                vector_visited_transpose.push_back(first_transpose_before_pa);
+            }
+        }
+    };
+
+    for (auto& op : model->get_ordered_ops()) {
+        if (ov::is_type<ov::op::PagedAttentionExtension>(op)) {
+            find_transpose_ops(op);
+        } else if (const auto& subgraph = ov::as_type_ptr<ov::hetero::op::DeviceSubgraph>(op)) {
+            for (auto& node : subgraph->get_function()->get_ordered_ops()) {
+                if (ov::is_type<ov::op::PagedAttentionExtension>(node)) {
+                    find_transpose_ops(node);
+                }
+            }
+        }
+    }
+
+    for (auto& node : vector_visited_transpose) {
+        std::map<size_t, NodePtr> org_users;
+        auto output_shape = node->get_output_partial_shape(0);
+        int num_key_value_heads = static_cast<int>(output_shape[2].get_length());
```

Comment:
maybe derive head info from page attention is better?

Reply:
I used a new method to fix this issue without adding any new nodes, please review again, thanks.